src/types.ts

```diff
@@ -33,33 +38,19 @@ export type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<
 
 /* Unknown Record */
 export type UR = Record<string, unknown>;
-export type UnknownType = UR; //alias to avoid breaking change
-
-export type DefaultGenerics = {
-  attachmentType: UR;
-  channelType: UR;
-  commandType: LiteralStringForUnion;
-  eventType: UR;
-  memberType: UR;
-  messageType: UR;
-  pollOptionType: UR;
-  pollType: UR;
-  reactionType: UR;
-  userType: UR;
-};
-
-export type ExtendableGenerics = {
-  attachmentType: UR;
-  channelType: UR;
-  commandType: string;
-  eventType: UR;
-  memberType: UR;
-  messageType: UR;
-  pollOptionType: UR;
-  pollType: UR;
-  reactionType: UR;
-  userType: UR;
-};
+export type UnknownType = UR; // alias to avoid breaking change
+
+export interface CustomAttachmentType {}
+export interface CustomChannelType {}
+export interface CustomCommandType {}
+export interface CustomEventType {}
+export interface CustomMemberType {}
+export interface CustomMessageType {}
+export interface CustomPollOptionType {}
+export interface CustomPollType {}
+export interface CustomReactionType {}
+export interface CustomUserType {}
+export interface CustomThreadType {}
```

Comment:
As for the naming - are we happy with this? I'd personally drop the `Type` suffix as it does not really fit (being an interface). It's breaking either way. Personally - I'd either replace `Type` suffix with `Data` suffix or just remove the suffix altogether.

Reply:
I like the `CustomXyzData` more. That's the term we use across our documentation.
Also, I think it would be good if we moved these to a `customTypes.ts` file. It should make it more obvious which types are meant to be extended.

Reply:
[Done](https://github.com/GetStream/stream-chat-js/pull/1458/commits/093493604020cd283f4ab5827f6480dedefcc0e7). :)